modules/programs/swaylock.nix

```diff
@@ -1,32 +1,44 @@
-{ config, lib, ... }:
+{ pkgs, config, lib, ... }:
+
+with lib;
 
 let cfg = config.programs.swaylock;
 in {
-  meta.maintainers = [ lib.hm.maintainers.rcerc ];
+  meta.maintainers = [ hm.maintainers.rcerc ];
+
+  options.programs.swaylock = {
+    enable = mkEnableOption "swaylock";
```

Comment:
We could add an assertion to ensure that if `programs.swaylock.settings != { }`, that `programs.swaylock.enable` must be `true`.

That would make the migration easier.

Reply:
Good idea, I will implement this. :)

Reply:
I'd suggest making it a waning instead of an error.

Reply:
I'm not quite sure how to exactly combine the combine these things. If enable isn't set, should we still write the config file in addition to printing the warning? Or should only the warning be printed and config isn't written?
  
  ```nix
  config = {
    home.packages = mkIf cfg.enable [ cfg.package ];

    warnings = mkIf (!cfg.enable && cfg.settings != { }) [''
      Warning: swaylock.settings is set but swaylock is disabled
    ''];

    xdg.configFile."swaylock/config" = mkIf (cfg.settings != { }) {
    # OR
    xdg.configFile."swaylock/config" = mkIf (cfg.enable && cfg.settings != { }) {
```

Reply:
@rycee @rcerc any opinion on this topic? I also wonder if there is a policy for the this in general. Should every program have an `enable`/`package` option? I guess this is what I would expect as a user...

Reply:
>  I do not see why someone would configure swaylock in Home Manager but not want it to generate a configuration file. (If the user would really like to disable it, they could reset programs.swaylock.settings with lib.mkForce { }.)

I understand your point, but isn't that the case for most/all home-manager programs? For what reason does the enable option then exist?

Reply:
I cannot answer that (maybe someone else can), but I am not against adding an `enable` option to the `swaylock` module.

Reply:
Sure, just wanted to understand... :)

Reply:
I'm personally of the opinion that an `enable` option can be good, is more in-line with the rest of the modules, and a minimal change. Same reasoning for `package`.

Reply:
Sorry for the delay. It is typically good to have an enable option, in some cases i've omitted it and I've usually have regretted it in hindsight. Personally I'm not fond of the package option but people seem very attached to them so I don't mind if they are added.

In this case I would suggest adding an enable option with the default set to something like

    versionOlder config.home.stateVersion "23.05"

so that old configurations that expect `settings != {}` to be sufficient will continue working. There should be no warning or assertion. The main point of having an enable option is to be able to unconditionally set the other module options and toggle the whole thing on or off with the enable option. See e.g. [my configuration of the gnome-terminal module](https://git.sr.ht/~rycee/configurations/tree/89ad940adc7ca74c60c76003dbee9ad12060c3fc/item/user/common.nix#L677-689).

In other words, if you add an enable option then setting it to false should result in no change to the user's environment. As if the user did not use the module at all.

Reply:
I've added the default as suggested, nice idea. I don't get why the tests are currently failing.